tests/core/plugins/widgets/test_raw_data_widget.py

```
@@ -0,0 +1,452 @@
+        self.assertIn("1.5", result)  # Float
+        self.assertIn("True", result)  # Boolean
+        self.assertIn("2023-01-01", result)  # Date
+
+    def test_render_exception_handling(self):
+        """Test exception handling during HTML generation."""
+        df = pd.DataFrame({"col1": ["A", "B", "C"], "col2": [1, 2, 3]})
+
+        params = RawDataWidgetParams()
+
+        # Mock to_html to raise an exception
+        with pytest.raises(Exception):
```

Ruff (0.11.9)

Do not assert blind exception: `Exception`

(B017)